src/components/ButtonWithMenu.js

```diff
@@ -0,0 +1,111 @@
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {View} from 'react-native';
+import _ from 'underscore';
+import styles from '../styles/styles';
+import Button from './Button';
+import ButtonWithDropdown from './ButtonWithDropdown';
+import PopoverMenu from './PopoverMenu';
+
+const propTypes = {
+    /** Text to display for the menu header */
+    menuHeaderText: PropTypes.string,
+
+    /** Callback to execute when the main button is pressed */
+    onPress: PropTypes.func,
+
+    /** Callback to execute when a menu item is selected */
+    onChange: PropTypes.func,
+
+    /** Whether we should show a loading state for the main button */
+    isLoading: PropTypes.bool,
+
+    /** Should the confirmation button be disabled? */
+    isDisabled: PropTypes.bool,
+
+    /** Menu options to display */
+    /** [{text: 'Pay with Expensify', icon: Wallet}, {text: 'PayPal', icon: PayPal}, {text: 'Venmo', icon: Venmo}] */
+    options: PropTypes.arrayOf(PropTypes.shape({
+        text: PropTypes.string.isRequired,
+        icon: PropTypes.elementType,
+        iconWidth: PropTypes.number,
```

Comment:
Looks like you're not using these parameters, I wonder if we should remove them since they should be standard anyway to keep the UI consistent

Reply:
These params would technically get used if passed in by the parent component. It's not obvious at all, but the spread `...` operator on line 94 would keep these options